multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/SyncFlowableStep.java

```diff
@@ -208,6 +211,31 @@ protected ProcessStepHelper getStepHelper() {
         return stepHelper;
     }
 
+    protected void processServiceCreationFailure(ProcessContext context, CloudServiceInstanceExtended serviceInstance,
+                                               CloudOperationException e) {
+        processServiceActionFailure(context, serviceInstance, e, MessageFormat.format(Messages.COULD_NOT_CREATE_OPTIONAL_SERVICE, serviceInstance.getName()));
+    }
+
+    protected void processServiceUpdateFailure(ProcessContext context, CloudServiceInstanceExtended serviceInstance,
```

Comment:
processService... methods don't really belong in this superclass. A more appropriate place would be ServiceStep or if that doesn't work the most concrete shared superclass between the steps. that require the methods